zephyr/src/supp_wrapper.c

```diff
@@ -0,0 +1,583 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/*
+ * WPA Supplicant / main() function for Zephyr OS
+ * Copyright (c) 2003-2013, Jouni Malinen <j@w1.fi>
+ *
+ * This software may be distributed under the terms of the BSD license.
+ * See README for more details.
+ */
+
+#include <logging/log.h>
+
+#include "supp_main.h"
+#include "includes.h"
+#include "common.h"
+#include "common/defs.h"
+#include "wpa_supplicant/config.h"
+#include "wpa_supplicant_i.h"
+#include "driver_i.h"
+#include "fst/fst.h"
+#include "p2p_supplicant.h"
+#include "wpa_supplicant_i.h"
+#include "wifi_mgmt.h"
+
+//#define MAX_SSID_LEN 32
+
+int cli_main (int, const char **);
+extern struct wpa_supplicant *wpa_s_0;
+
+int zephyr_supp_disable_network(int id)
+{
+    char cmd[512] = {'\0'};
+    const char *argv[CONFIG_SHELL_ARGC_MAX + 1]; /* +1 reserved for NULL */
+    size_t argc = 0;
+    const char **argvp;
+    char quote;
+    int ret = -1;
+
+    os_snprintf(cmd, sizeof(cmd), "disable_network %d", id);
+    argv[0] = cmd;
+    argvp = &argv[0];
+
+    quote = z_shell_make_argv(&argc, argvp, cmd, 4);
+
+	if (argc == 0) {
+		return -ENOEXEC;
+	} else if ((argc == 1) && (quote != 0)) {
+		wpa_printf(MSG_ERROR, "not terminated: %c\n", quote);
+		return -ENOEXEC;
+	}
+
+    ret = cli_main (argc, argv);
```

Comment:
The space after `cli_main ` don't seem to be intended? There are other places like this as well.